src/views/domain-workflows/domain-workflows-table/domain-workflows-table.tsx

```diff
@@ -66,23 +66,33 @@ export default function DomainWorkflowsTable(props: Props) {
     },
   });
 
-  const workflows = useMemo(
-    () => data.pages.flatMap((page) => page.workflows ?? []),
-    [data]
-  );
+  if (isLoading) {
+    return <SectionLoadingIndicator />;
+  }
+
+  const workflows = data?.pages.flatMap((page) => page.workflows ?? []) ?? [];
+
+  if (workflows.length === 0) {
+    const errorPanelProps = getWorkflowsErrorPanelProps({
+      error,
+      areSearchParamsAbsent:
+        !queryParams.search &&
+        !queryParams.status &&
+        !queryParams.timeRangeStart &&
+        !queryParams.timeRangeEnd,
+    });
 
-  if (
-    !queryParams.search &&
-    !queryParams.status &&
-    !queryParams.timeRangeStart &&
-    !queryParams.timeRangeEnd &&
-    workflows.length === 0
-  ) {
-    throw new Error(NO_WORKFLOWS_ERROR_MESSAGE);
+    if (errorPanelProps) {
+      return (
+        <styled.PageSection>
+          <ErrorPanel {...errorPanelProps} reset={refetch} />
+        </styled.PageSection>
+      );
+    }
   }
 
   return (
-    <PageSection>
+    <styled.PageSection>
```

Comment:
maybe change the type of the page sectio element here to a div. 
And add another section component arround the search plus the table

Reply:
I moved the PageSection wrappers one level higher to the DomainPageContent component